File: BookManagerAPI/BookManagerAPI/Models/Book.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BookManagerAPI.Models
{
    public class Book
    {
        [Key]
        public int Id { get; set; }
        [Required]
        [Column(TypeName ="nvarchar(100)")]
        public string Title { get; set; } = "";
        [Required]
        [Column(TypeName = "nvarchar(100)")]
        public string Author { get; set; } = "";
        [Required]
        [Column(TypeName = "nvarchar(100)")]
        [RegularExpression(@"^\d{3}-\d{10}$")]
        public string ISBN { get; set; } = "";

        [Required]
        public DateTime PublicationDate { get; set; }
    }
}

File: BookManagerApp/src/app/models/book.ts
export class Book {

    id: number;
    title: string;
    author: string;
    isbn: string;
    publicationDate: Date;

    constructor( id: number = 0,
    title: string = '',
    author: string = '',
    isbn: string = '',
    publicationDate: Date = new Date()) {
        this.id = id;
        this.title = title;
        this.author = author;
        this.isbn = isbn;
        this.publicationDate = publicationDate;
    }

    
}
